test/FhirLoader.Tests.E2E/FHIRLoaderE2ETest.cs

```diff
@@ -93,5 +94,152 @@ public async void BlobFhir_ndjson_Test()
+        /// </summary>
+        [Fact]
+        public async void LocalFhir_ifpassingwrongpath_Test()
+        {
+            CommandOptions commandOptions = new()
+            {
+                FhirUrl = _config.FhirURL,
+                PackagePath = @"../../../TestData/package/example",
+                BatchSize = _config.Batchsize,//Must be betweeen 1 & 500,Default 500
+                Concurrency = _config.Concurrency//Must be betweeen 1 & 50,Default 50
+            };
+            int response = await Program.Run(commandOptions);
+            Action throwingAction = () =>
+            {
+                throw new ArgumentException();
+            };
+            Assert.Throws<ArgumentException>(throwingAction);
+
+
+        }
+
+        /// <summary>
+        /// Test Case for Package Files.
+        /// By passing Package Path,it will read all the files inside the package folder and process them to the Fhir URL.
+        /// </summary>
+        [Fact]
+        public async void LocalFhir_package_Test()
+        {
+            CommandOptions commandOptions = new()
+            {
+                FhirUrl = _config.FhirURL,
+                PackagePath = @"../../../TestData/package",
+                BatchSize = _config.Batchsize,//Must be betweeen 1 & 500,Default 500
+                Concurrency = _config.Concurrency//Must be betweeen 1 & 50,Default 50
+            };
+            int response = await Program.Run(commandOptions);
+            Assert.Equal(0, response);
+        }
+
+        /// <summary>
+        /// Check that the json file has the resource type bundle.
+        /// If a file without a bundle resource type exists in a bundle folder, then the tool will skip those files.
+        /// </summary>
+        [Fact]
+        public void CheckIfFileswithoutbundleresourceTypeExist_Test()
+        {
+            var inputBundles = Directory
+               .EnumerateFiles(@"../../../TestData/bundle", "*", SearchOption.AllDirectories)
+               .Where(s => s.EndsWith(".json"));
+            int i = 0;
+            foreach (var filePath in inputBundles.OrderBy(x => x))
+            {
+                if (filePath.EndsWith(".json"))
+                {
+                    JObject data = JObject.Parse(File.ReadAllText(filePath));
+                    if (data.GetValue("resourceType")?.Value<string>()?.ToLower() != "bundle")
+                    {
+                        i++;
+                    }
+                }
+            }
+            if (i > 0)
+            {
+                Assert.Fail("Files that do not have a bundle resource type will be skipped.");
+            }
+
+        }
+
+        /// <summary>
+        /// Test case to check type in package json.
+        /// Check to see if the type in package.json matches values such as Conformance, fhir.ig, and Core.�
+        /// returns true if matched, otherwise false.
+        /// </summary>
+        [Fact]
+        public void checkIfPackageTypeExists_Test()
```

Comment:
What exactly is this testing? I don't see any execution of the program code.

Reply:
I have changed the logic for this case. Instead of a static value, it will read the file from the folder and check if the type in the package json does not match values such as conformance, fhir.ig, and core, then throw an exception.